list_resources.sh

```diff
@@ -1,19 +1,23 @@
-#!/bin/bash
-specs_dir="./specs"
+#!/bin/sh
 
-function findResourceDir(){
-    for file in `ls $1`
-    do
-    	work_dir=$1"/"$file
-        if [ -d $work_dir ]
-        then
-        	if [ -n "$(git status -s $work_dir)" ] && [ -f $work_dir"/api.yaml" ] \
-        		&& ([ -f $work_dir"/ansible.yaml" ] || [ -f $work_dir"/terraform.yaml" ]); then
-				echo ${work_dir:8}		# remove the directory prefix './specs/'
+findResourceDir(){
+	for file in `ls $1`
+	do
+		work_dir=$1"/"$file
+		if [ -d $work_dir ]
+		then
+			if [ -n "$(git status -s $work_dir)" ] && [ -f $work_dir"/api.yaml" ] \
+				&& ([ -f $work_dir"/ansible.yaml" ] || [ -f $work_dir"/terraform.yaml" ]);
+			then
+				# remove the directory prefix './specs'
+				expr substr "$work_dir" 9 "${#work_dir}"
 			fi
-            findResourceDir $work_dir	# find resource directory recursively
-        fi
-    done
+			# find resource directory recursively
+			findResourceDir $work_dir
+		fi
+	done
 }
-
+# go to magic-module-specs directory
+cd `dirname $0`
+specs_dir='./specs'
```

Comment:
Please get absolute path of `specs` then pass it your function. Don't change current directory and use relative directory.

Reply:
Git command needs to work in the repo directory. What's more, it's in a new shell script, and it won't change the work directory of the caller.

Reply:
Why not directly set the specs_dir to be `dirname $0`/specs ?

Reply:
Because git needs to run in the root directory of magic-modules-spec.